### PR TITLE
Table widget

### PR DESCRIPTION
**Pull request by MTrost**
- [x] Make :field-key OR :header-cell mandatory, to show column (edit: rows is mandatory, everything else optional)
- [x] Make :rows or :db-path mandatory, to pass row data down OR get them from db (currently data is passed down in :rows, a db-path is used in sort-config)
- [x] Implement col sorting in widget and edges
- [x] add custom sort-field functionality for deployment
- [x] implement multi col sorting

### CODE REVIEW

**Comment by MTrost** [comment]
Solves #1005

File: code/src/cljs/sixsq/nuvla/ui/plugins/table.cljs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+     :fx [[:dispatch [fetch-event]]]}))
+
+(reg-sub
+  ::sort-direction
+  (fn [db [_ db-path sort-key]]
+    (let [ordering (get db db-path)]
+      (some #(when (= sort-key (first %)) (second %)) ordering))))
+
+(defn Sort [{:keys [db-path field-key sort-key fetch-event]
+             :or {sort-key field-key}}]
```
Comment (0xbase12) [inline code/src/cljs/sixsq/nuvla/ui/plugins/table.cljs:73]: :or map can't use destructured fields of the map itself

**Review by 0xbase12 — CHANGES_REQUESTED** [review]
Getting blank pages when trying to load edges or deployments pages

**Comment by 0xbase12** [comment]
The multi column sort is interesting and well done, but don't know if it's too complicated from UX perspective to our end users.
I propose to ask the team what they think during a demo meeting. 

We can have an option that ask to sort only by one column.

**Comment by MTrost** [comment]
I was thinking about adding this option but then decided against it, because it would make it even more complicated with additional options.

The multi-column sort currently implemented works the same as in Excel or Google Sheets and isn't even that different from single-column sort, because the last clicked column gets full sort applied.

But it wouldn't be that difficult to make it optional and/or configurable

**Pull request merged**